### PR TITLE
Implement Twilio for booking reminders

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR uses Twilio to send booking reminders to attendees. The reminders are set by the organizer. 

Addresses #1938 & #1780 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Testing SMS Reminders
1. Signup for a Twilio account
2. Create a number (Should be able to do so on homepage)
3. Add the the account id and auth token to .env
4. Create a messaging service and add the created number
5. Add the messaging_sid to the .env
6. Create a new event type, in the advanced options add an SMS reminder
7. Create a new booking, note that the scheduled send date should be within 60mins of the current time and <7 days
8. Go to the Twilio portal and go to messaging overview, you should see the message scheduled
	- Optionally you can also comment out the sendAt parameter to get the text message immediately 
9. Reschedule the booking, in the DB under attendeeReminder you should see the updated scheduled date
10. Cancel the booking, you should see the reminder is no longer in the DB and on the Twilio console the message should say cancelled

### Testing Email reminders
1. Signup for a Sendgrid account
2. Get an API key and setup the approved sender email
3. Add the API key and sender email to .env
4. Create a new event type, in the advanced options add an email reminder
5. Schedule a booking under that event type that is within 60 mins
6. You should receive a reminder email
7. Schedule another booking that is 1 day in advance.
    - If you do not receive an error booking the event then the email is scheduled

### CODE REVIEW

File: apps/web/lib/events/EventManager.ts [hunk 1]
```diff
@@ -238,14 +239,20 @@ export default class EventManager {
       },
     });
 
+    const reminderDeletes = prisma.attendeeReminder.deleteMany({
+      where: {
+        bookingUid: booking.uid,
+      },
+    });
+
```
Comment (joeauyeung) [inline apps/web/lib/events/EventManager.ts:247]: To delete a booking we have to delete all other records connected to it, this deletes any reminders that are associated with that booking

File: apps/web/lib/reminders/smsReminderManager.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          sendTo: reminderPhone,
+          referenceId: response.sid,
+          scheduledDate: dayjs().toDate(),
+          scheduled: true,
+        },
+      });
+    } catch (error) {
+      console.log(`Error sending SMS with error ${error}`);
+    }
+  }
```
Comment (joeauyeung) [inline apps/web/lib/reminders/smsReminderManager.tsx:59]: If the scheduled date for the reminder is within 60 minutes of the current time then send a reminder immediately to the attendee. The current limitation of Twilio scheduling SMS messages is that it has to be 60 minutes in the future.

File: apps/web/lib/reminders/smsReminderManager.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          sendTo: reminderPhone,
+          referenceId: response.sid,
+          scheduledDate: scheduledDate.toDate(),
+          scheduled: true,
+        },
+      });
+    } catch (error) {
+      console.log(`Error scheduling SMS with error ${error}`);
+    }
+  }
```
Comment (joeauyeung) [inline apps/web/lib/reminders/smsReminderManager.tsx:88]: If the scheduled date is within 7 days then schedule the SMS reminder to be sent. The current Twilio limitation is that SMS messages can not be scheduled further than 7 days.

File: apps/web/lib/reminders/smsReminderManager.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        },
+        method: "SMS",
+        sendTo: reminderPhone,
+        referenceId: "",
+        scheduledDate: scheduledDate.toDate(),
+        scheduled: false,
+      },
+    });
+  }
+};
```
Comment (joeauyeung) [inline apps/web/lib/reminders/smsReminderManager.tsx:107]: If the scheduled date is further than 7 days then write the reminder to the db to be scheduled later in the cron job.

File: apps/web/lib/reminders/smsReminderManager.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+    await prisma.attendeeReminder.delete({
+      where: {
+        referenceId: referenceId,
+      },
+    });
+  } catch (error) {
+    console.log(`Error canceling reminder with error ${error}`);
+  }
+};
```
Comment (joeauyeung) [inline apps/web/lib/reminders/smsReminderManager.tsx:122]: If we want to delete scheduled messages we have to do them one at a time. Also when updating bookings we can not update the scheduled message directly. We have to delete and create a new one.

File: apps/web/lib/reminders/templates/reminderTemplate.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  title: string,
+  organizerName: string,
+  startTime: string,
+  attendeeTimeZone: string
+) => {
+  return `Hi, this is a reminder that you have a ${title} with 
+  ${organizerName} at ${dayjs(startTime).format("YYYY MMMM DD h:mmA")} ${attendeeTimeZone}`;
+};
+
+export default reminderTemplate;
```
Comment (joeauyeung) [inline apps/web/lib/reminders/templates/reminderTemplate.ts:15]: Template for SMS reminder

File: apps/web/lib/types/booking.ts [hunk 7]
```diff
@@ -23,6 +23,7 @@ export type BookingCreateBody = {
   user?: string | string[];
   language: string;
   customInputs: { label: string; value: string }[];
+  reminderPhone: string;
```
Comment (joeauyeung) [inline apps/web/lib/types/booking.ts:26]: This type is used when creating a booking

File: apps/web/pages/api/book/event.ts [hunk 8]
```diff
@@ -555,6 +558,13 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       }
 
       await sendRescheduledEmails({ ...evt, additionInformation: metadata });
+
+      // Reschedule reminders
+      for (const reminder of eventType.attendeeReminders) {
+        if (reminder.method === "SMS" && evt.reminderPhone) {
+          await scheduleSMSAttendeeReminder(evt, evt.reminderPhone, reminder);
+        }
+      }
```
Comment (joeauyeung) [inline apps/web/pages/api/book/event.ts:567]: When rescheduling bookings, we can not update the scheduled message directly. We have to delete them and create new ones.

File: apps/web/pages/api/book/event.ts [hunk 9]
```diff
@@ -582,6 +592,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         metadata.entryPoints = results[0].createdEvent?.entryPoints;
       }
       await sendScheduledEmails({ ...evt, additionInformation: metadata });
+
+      for (const reminder of eventType.attendeeReminders) {
+        if (reminder.method === "SMS" && evt.reminderPhone) {
+          await scheduleSMSAttendeeReminder(evt, evt.reminderPhone, reminder);
+        }
+      }
```
Comment (joeauyeung) [inline apps/web/pages/api/book/event.ts:600]: If the reminder type is SMS we have to schedule the reminders one at a time.

File: apps/web/pages/api/cron/scheduleReminders.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  if (process.env.CRON_API_KEY !== apiKey) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+
+  const unscheduledReminders = await prisma.attendeeReminder.findMany({
+    where: {
+      scheduled: false,
+    },
+  });
```
Comment (joeauyeung) [inline apps/web/pages/api/cron/scheduleReminders.ts:27]: Look for any reminders that are not scheduled

File: apps/web/pages/api/cron/scheduleReminders.ts [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+  if (!unscheduledReminders.length) res.json({ ok: true });
+
+  const inSevenDays = dayjs().add(7, "day");
+
+  for (const reminder of unscheduledReminders) {
+    if (dayjs(reminder.scheduledDate).isBefore(inSevenDays)) {
```
Comment (joeauyeung) [inline apps/web/pages/api/cron/scheduleReminders.ts:34]: Checks to see if the schedule date for the reminder is within the 7 day window

File: apps/web/pages/api/cron/scheduleReminders.ts [hunk 12]
```diff
@@ -0,0 +1,74 @@
+/* Schedule any attendee reminder that falls within 7 days for SMS 
+and 72 hours for email */
```
Comment (joeauyeung) [inline apps/web/pages/api/cron/scheduleReminders.ts:2]: This is the endpoint for the cron job to schedule any reminders that fall within the 7 day window for SMS

File: apps/web/pages/event-types/[type].tsx [hunk 13]
```diff
@@ -298,6 +313,12 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
     setCustomInputs([...customInputs]);
   };
 
+  const removeReminder = (index: number) => {
+    formMethods.getValues("attendeeReminders").splice(index, 1);
+    attendeeReminders.splice(index, 1);
+    setAttendeeReminders([...attendeeReminders]);
+  };
+
```
Comment (joeauyeung) [inline apps/web/pages/event-types/[type].tsx:321]: Removes any reminders from the form

File: apps/web/pages/event-types/[type].tsx [hunk 14]
```diff
@@ -1334,6 +1356,72 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
+                                          {t("edit")}
+                                        </Button>
+                                        <button type="button" onClick={() => removeReminder(idx)}>
+                                          <XIcon className="h-6 w-6 border-l-2 pl-1 hover:text-red-500 " />
+                                        </button>
+                                      </div>
+                                    </div>
+                                  </li>
+                                )
+                              )}
```
Comment (joeauyeung) [inline apps/web/pages/event-types/[type].tsx:1408]: Renders current reminders on the frontend

File: apps/web/pages/event-types/[type].tsx [hunk 15]
```diff
@@ -1334,6 +1356,72 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
+                                  }}
+                                  color="secondary"
+                                  type="button"
+                                  StartIcon={PlusIcon}>
+                                  {t("add_reminder")}
+                                </Button>
+                              </li>
+                            </ul>
+                          </div>
+                        </div>
```
Comment (joeauyeung) [inline apps/web/pages/event-types/[type].tsx:1423]: Button to add new reminders

File: apps/web/pages/event-types/[type].tsx [hunk 16]
```diff
@@ -1634,6 +1722,64 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
+                      }}
+                      onCancel={() => {
+                        setSelectedAttendeeReminderModalOpen(false);
+                      }}
+                    />
+                  </div>
+                </DialogContent>
+              </Dialog>
+            )}
+          />
```
Comment (joeauyeung) [inline apps/web/pages/event-types/[type].tsx:1782]: Modal to add or edit new reminders

File: apps/web/server/routers/viewer/eventTypes.tsx [hunk 17]
```diff
@@ -63,6 +70,42 @@ function handleCustomInputs(customInputs: EventTypeCustomInput[], eventTypeId: n
...
+        id: { in: remindersIdsToDelete },
+      },
+    },
+    createMany: {
+      data: remindersToCreate,
+    },
+    update: remindersToUpdate,
+  };
+}
+
```
Comment (joeauyeung) [inline apps/web/server/routers/viewer/eventTypes.tsx:108]: Writes the event type reminders to the DB

File: packages/prisma/schema.prisma [hunk 18]
```diff
@@ -303,6 +305,38 @@ model EventTypeCustomInput {
...
+}
+
+model EventTypeAttendeeReminder {
+  id          Int                               @id @default(autoincrement())
+  eventTypeId Int
+  eventType   EventType                         @relation(fields: [eventTypeId], references: [id])
+  method      EventTypeAttendeeReminderMethod
+  time        Int
+  unitTime    EventTypeAttendeeReminderUnitTime
+}
```
Comment (joeauyeung) [inline packages/prisma/schema.prisma:326]: The reminders associated to the event type

File: packages/prisma/schema.prisma [hunk 19]
```diff
@@ -303,6 +305,38 @@ model EventTypeCustomInput {
...
+  id            Int      @id @default(autoincrement())
+  // At the time of creting the event the db id isn't created but the uid is
+  bookingUid    String
+  booking       Booking? @relation(fields: [bookingUid], references: [uid])
+  method        String
+  sendTo        String
+  referenceId   String   @unique
+  scheduledDate DateTime
+  scheduled     Boolean
+}
```
Comment (joeauyeung) [inline packages/prisma/schema.prisma:338]: The reminder associated to the booking

File: apps/web/ee/pages/api/cron/scheduleEmailReminder.ts [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+
+      try {
+        const response = await sgMail.send({
+          to: booking.attendees[0].email,
+          from: "j.auyeung419@gmail.com",
+          subject: "Test email",
+          content: [
+            {
+              type: "text/html",
+              value: new reminderTemplate(booking, booking.attendees[0]).getHtmlBody(),
```
Comment (joeauyeung) [inline apps/web/ee/pages/api/cron/scheduleEmailReminder.ts:64]: Getting type error here since booking is not a CalendarEvent type but all the necessary information is being passed

File: apps/web/ee/pages/api/cron/scheduleSMSReminders.ts [hunk 21]
```diff
@@ -0,0 +1,76 @@
+/* Schedule any attendee reminder that falls within 7 days for SMS */
```
Comment (joeauyeung) [inline apps/web/ee/pages/api/cron/scheduleSMSReminders.ts:1]: Cron job to schedule SMS reminders that are within 7 days

File: apps/web/pages/api/book/event.ts [hunk 22]
```diff
@@ -555,6 +562,16 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+
+      // Reschedule reminders
+      for (const reminder of eventType.attendeeReminders) {
+        if (reminder.method === "SMS" && evt.reminderPhone) {
+          await scheduleSMSAttendeeReminder(evt, evt.reminderPhone, reminder);
+        }
+        if (reminder.method === "EMAIL" && evt.reminderPhone) {
+          await scheduleEmailReminder(evt, evt.attendees[0].email, reminder);
+        }
+      }
```
Comment (joeauyeung) [inline apps/web/pages/api/book/event.ts:574]: The event manager deals with deleting the previous scheduled reminders. We can then create new ones

File: apps/web/pages/event-types/[type].tsx [hunk 23]
```diff
@@ -1334,6 +1405,76 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
                           </div>
                         </div>
 
+                        {(eventType.users[0].plan === "TRIAL" || eventType.users[0].plan === "PRO") && (
```
Comment (joeauyeung) [inline apps/web/pages/event-types/[type].tsx:1408]: Check to see if a user is eligible for attendee reminders

**Review by joeauyeung — COMMENTED** [review]
Some files from main were added to files changed. I commented on the files that I worked on

File: apps/web/ee/lib/reminders/emailReminderManager.ts [hunk 24]
```diff
@@ -0,0 +1,143 @@
...
+client.setApiKey(sendgridAPIKey);
+
+export const scheduleEmailReminder = async (
+  evt: CalendarEvent,
+  email: string,
+  attendeeReminder: EventTypeAttendeeReminder
+) => {
+  const { startTime } = evt;
+  const uid = evt.uid as string;
+  const currentDate = dayjs();
```
Comment (alishaz-polymath) [inline apps/web/ee/lib/reminders/emailReminderManager.ts:26]: Shouldn't we use UTC here?

**Review by alishaz-polymath — CHANGES_REQUESTED** [review]
This is a pretty large PR and looks mostly good, but I have added a few comments as a starting point. Will block the merge for now just in case because the migration file is not okay right now. Will continue testing it in terms of functionality and see how that goes.

File: packages/prisma/migrations/20220317195138_add_attendee_reminders/migration.sql [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+CREATE UNIQUE INDEX "AttendeeReminder_referenceId_key" ON "AttendeeReminder"("referenceId");
+
+-- AddForeignKey
+ALTER TABLE "EventTypeAttendeeReminder" ADD CONSTRAINT "EventTypeAttendeeReminder_eventTypeId_fkey" FOREIGN KEY ("eventTypeId") REFERENCES "EventType"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+-- AddForeignKey
+ALTER TABLE "AttendeeReminder" ADD CONSTRAINT "AttendeeReminder_bookingUid_fkey" FOREIGN KEY ("bookingUid") REFERENCES "Booking"("uid") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+-- RenameIndex
+ALTER INDEX "Booking.uid_unique" RENAME TO "Booking_uid_key";
```
Comment (alishaz-polymath) [inline packages/prisma/migrations/20220317195138_add_attendee_reminders/migration.sql:41]: All the rename index migration calls need to be removed

File: apps/web/ee/lib/reminders/emailReminderManager.ts (hunk 24 above)
Comment (alishaz-polymath) [reply apps/web/ee/lib/reminders/emailReminderManager.ts:26]: Hm. I think it is always best practice to standardise UTC to cover the possibilities of using the services without any issue when moving around different timezones. I'm not entirely sure yet if that would affect the functionality of the PR at all, but it's best to think over it.

**Comment by alishaz-polymath** [comment]
Been trying to test it out with Sendgrid but for some reason I am not even able to verify my account there (perhaps an issue on their end). This is working as a blocker for me to test it out. If someone else can test out the functionality, that would be great. I'll review the code until then

**Comment by alishaz-polymath** [comment]
Just tried testing email reminder and here's the error I face @joeauyeung . Any idea why this is and what can be done to get it sorted so that I can continue testing?
image

I have tested booking on main branch and it works fine, so the error is linked to the changes in this branch.

**Comment by joeauyeung** [comment]
> Just tried testing email reminder and here's the error I face @joeauyeung . Any idea why this is and what can be done to get it sorted so that I can continue testing? image
> 
> I have tested booking on main branch and it works fine, so the error is linked to the changes in this branch.

This is really weird, I experienced this early in development but then I just recreated the branch and haven't run into it since. I'll take a look at it.

File: packages/prisma/schema.prisma [hunk 26]
```diff
@@ -3,7 +3,7 @@
 
 datasource db {
   provider = "postgresql"
-  url      = env("DATABASE_URL")
+  url      = "postgresql://postgres:@localhost:5450/calendso"
```
Comment (zomars) [inline packages/prisma/schema.prisma:6]: Let's undo this change

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 27]
```diff
@@ -466,7 +468,26 @@ const BookingPage = ({ eventType, booking, profile }: BookingPageProps) => {
                         )}
                       </div>
                     ))}
-                  {!eventType.disableGuests && (
+                  {eventType.attendeeReminders.some((reminder) => reminder.method === "SMS") && (
```
Comment (alannnc) [inline apps/web/components/booking/pages/BookingPage.tsx:471]: Can we use an Enum for "SMS"

File: apps/web/ee/lib/reminders/templates/reminderSMSTemplate.ts [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+
+import { CalendarEvent } from "@lib/integrations/calendar/interfaces/Calendar";
+
+const reminderTemplate = (
+  title: string,
+  organizerName: string,
+  startTime: string,
+  attendeeTimeZone: string
+) => {
+  return `Hi, this is a reminder that you have a ${title} with
```
Comment (alannnc) [inline apps/web/ee/lib/reminders/templates/reminderSMSTemplate.ts:11]: SMS shouldn't be more than 140 chars, we should valid that and short if needed.

File: apps/web/ee/pages/api/cron/scheduleSMSReminders.ts [hunk 29]
```diff
@@ -0,0 +1,84 @@
...
+          where: {
+            id: reminder.id,
+          },
+          data: {
+            scheduled: true,
+            referenceId: response.sid,
+          },
+        });
+      } catch (error) {
+        console.log(`Error scheduling SMS with error ${error}`);
```
Comment (alannnc) [inline apps/web/ee/pages/api/cron/scheduleSMSReminders.ts:80]: Add a // @TODO: Report to sentry

**Review by alannnc — COMMENTED** [review]
Looks good! Probably we need to separate logic between smsReminderManager and twilio, in this case you would have smsProvider and smsReminderManager. Just in case you want to switch from sms provider, or twilio doesn't send sms on specific country.

**Comment by alishaz-polymath** [comment]
Agree with @alannnc here

> Looks good! Probably we need to separate logic between smsReminderManager and twilio, in this case you would have smsProvider and smsReminderManager. Just in case you want to switch from sms provider, or twilio doesn't send sms on specific country.

**Review by alannnc — COMMENTED** [review]
I submitted this form, but thought 30 was the default value, then after saving it looked like 0 Minutes before the event. Probably leaving placeholder as default value will make it more consistent.
image

Sometimes changing country on phone resets my input, when having phone value and changing from international to another country,
image

When creating attendeeReminder I got this as response, it looks referenceId is being saved as `String` instead of response value.
Code: `referenceId: response.sid,` <-- sid exists?
image

After all sms was sent successfully, it seems I ran out of credit on sendgrid and can't bring my account back.

File: apps/web/components/ui/form/PhoneInput.tsx [hunk 30]
```diff
@@ -15,6 +15,7 @@ type PhoneInputProps = {
 export const PhoneInput = ({ control, name, ...rest }: Props<PhoneInputProps>) => (
   <BasePhoneInput
     {...rest}
+    international
```
Comment (joeauyeung) [inline apps/web/components/ui/form/PhoneInput.tsx:18]: When someone changes their country it still wipes their input but it now starts with the country code

File: apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx [hunk 31]
```diff
@@ -62,7 +61,7 @@ const CustomInputTypeForm: FC<Props> = (props) => {
           render={({ field }) => (
             <Select
               id="method"
-              defaultValue={selectedMethodOption}
+              defaultValue={selectedAttendeeReminder ? selectedMethodOption : methodOptions[0].value}
```
Comment (joeauyeung) [inline apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx:64]: Default reminder is SMS

File: apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx [hunk 32]
```diff
@@ -81,9 +80,8 @@ const CustomInputTypeForm: FC<Props> = (props) => {
         <div className="inline-flex w-full space-x-2 ltr:ml-2 rtl:mr-2 rtl:space-x-reverse">
           <input
             type="number"
-            className="focus:border-primary-500 focus:ring-primary-500 block w-12 rounded-sm border-gray-300 shadow-sm [appearance:textfield] ltr:mr-2 rtl:ml-2 sm:text-sm"
-            placeholder="30"
-            defaultValue={selectedAttendeeReminder?.time}
+            className="focus:border-primary-500 focus:ring-primary-500 block w-14 rounded-sm border-gray-300 shadow-sm [appearance:textfield] ltr:mr-2 rtl:ml-2 sm:text-sm"
+            defaultValue={selectedAttendeeReminder ? selectedAttendeeReminder.time : 30}
```
Comment (joeauyeung) [inline apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx:84]: Default reminder will be 30 minutes before the event

File: apps/web/ee/lib/reminders/smsProviders/twilioProvider.ts [hunk 33]
```diff
@@ -19,7 +19,7 @@ export const sendSMS = async (phoneNumber: string, body: string) => {
     to: phoneNumber,
   });
 
-  return response.sid;
+  return response;
```
Comment (joeauyeung) [inline apps/web/ee/lib/reminders/smsProviders/twilioProvider.ts:22]: Return the response object when sending a SMS

**Review by joeauyeung — COMMENTED** [review]
@alannnc addressing your comments

File: apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx (hunk 31 above)
Comment (zomars) [reply apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx:64]: We should not show this options if TWILIO keys aren't set

Comment (zomars) [reply apps/web/ee/components/eventtype/AttendeeReminderTypeForm.tsx:61]: Updated some changes about this

**Comment by PeerRich** [comment]
image
@joeauyeung some merge conflicts 🙌

**Comment by lharries** [comment]
Great work on this feature @joeauyeung! Do you have a rough eta for completion? I'm working out if we need to implement this ourselves our if this PR will fix it - thank you!

File: apps/web/pages/event-types/[type].tsx [hunk 34]
```diff
@@ -255,6 +264,13 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
...
+  const [selectedAttendeeReminder, setSelectedAttendeeReminder] = useState<
+    EventTypeAttendeeReminder | undefined
+  >(undefined);
+  const [selectedAttendeeReminderModalOpen, setSelectedAttendeeReminderModalOpen] = useState(false);
+  const [attendeeReminders, setAttendeeReminders] = useState<EventTypeAttendeeReminder[]>(
+    eventType.attendeeReminders.sort((a, b) => a.id - b.id) || []
+  );
 
   const periodType =
     PERIOD_TYPES.find((s) => s.type === eventType.periodType) ||
```
Comment (leog) [inline apps/web/pages/event-types/[type].tsx:276]: Another warning to take care of :)

**Review by leog — COMMENTED** [review]
Looks good, couldn't test it locally, not in my computer at the moment

**Review by zomars — CHANGES_REQUESTED** [review]
Code got outdated pretty fast, will re-draft until properly updated.
